File: python-code/ex11/funcionario_horista.py
from funcionario import Funcionario

class FuncionarioHorista(Funcionario):
  def __init__(self, nome, horas_trabalhadas, valor_hora):
    super().__init__(nome)
    self.horas_trabalhadas = horas_trabalhadas
    self.valor_hora = valor_hora

  def calcular_salario(self):
    return self.horas_trabalhadas * self.valor_hora

File: golang-code/ex11/funcionario_horista.go
package main

type FuncionarioHorista struct {
	Nome             string
	HorasTrabalhadas float64
	ValorHora        float64
}

func (f FuncionarioHorista) CalcularSalario() float64 {
	return f.HorasTrabalhadas * f.ValorHora
}